src/results/DownloadContainer.tsx

```diff
@@ -0,0 +1,168 @@
+    contentType: '',
+    data: '',
+  });
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const url = getDownloadUrl({
+      query,
+      columns: selectedColumns,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+      fileFormat,
+      compressed,
+      selectedAccessions: downloadAll ? [] : selectedEntries,
+    });
+    const link = document.createElement('a');
+    link.href = url;
+    link.target = '_blank';
+    link.setAttribute('download', '');
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+    history.goBack();
+  };
+
+  const handleCancel = () => {
+    history.goBack();
+  };
+
+  const handlePreview = (nPreview: number) => {
+    const fileFormatExcelReplaced = replaceExcelWithTsv(fileFormat);
+    const url = getDownloadUrl({
+      query,
+      columns: selectedColumns,
+      selectedFacets,
+      sortColumn,
+      sortDirection,
+      fileFormat: fileFormatExcelReplaced,
+      compressed,
+      size: nPreview,
+      selectedAccessions: downloadAll ? [] : selectedEntries,
+    });
+    setLoadingPreview(true);
+    fetchData(url, {
+      Accept: fileFormatToContentType.get(fileFormatExcelReplaced),
+    })
+      .then(response => {
+        const contentType = idx(
+          response,
+          o => o.headers['content-type']
+        ) as FileFormat;
+        setPreview({
+          data:
+            contentType === fileFormatToContentType.get(FileFormat.json)
+              ? JSON.stringify(response.data, null, 2)
+              : response.data,
+          url: idx(response, o => o.config.url) || '',
+          contentType,
+        });
+      })
+      .catch(e => {
+        // eslint-disable-next-line no-console
+        console.error(e);
+      })
+      .finally(() => {
+        setLoadingPreview(false);
+      });
+  };
+  const downloadUrl = getDownloadUrl({
+    query,
+    columns: selectedColumns,
+    selectedFacets,
+    sortColumn,
+    sortDirection,
+    fileFormat,
+    compressed,
+    selectedAccessions: downloadAll ? [] : selectedEntries,
+  });
+  return (
+    <DownloadView
+      onSubmit={handleSubmit}
+      onCancel={handleCancel}
+      onPreview={handlePreview}
+      selectedColumns={selectedColumns}
+      downloadAll={downloadAll}
+      fileFormat={fileFormat}
+      compressed={compressed}
+      onSelectedColumnsChange={setSelectedColumns}
+      nSelectedEntries={selectedEntries.length}
+      onDownloadAllChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+        setDownloadAll(e.target.value === 'true')
+      }
+      onFileFormatChange={(e: React.ChangeEvent<HTMLSelectElement>) =>
+        setFileFormat(e.target.value as FileFormat)
+      }
+      onCompressedChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+        setCompressed(e.target.value === 'true')
+      }
+      preview={
+        compareDownloadUrlsDisregardSize(preview.url, downloadUrl) &&
```

Comment:
For clarity, it would be nice to have this evaluated into a variable, maybe `resetPreview`.

Reply:
Yes, would be much clearer.